src/LanguageServer/Test/DiagnosticsTests.cs

```diff
@@ -116,131 +117,154 @@ public void TestInitialize()
 
-            var uri = doc.Uri;
-            var callReceived = false;
             var clientApp = Services.GetService<IClientApplication>();
-            var expectedSeverity = DiagnosticSeverity.Error;
+            var reported = new List<PublishDiagnosticsParams>();
             clientApp.When(x => x.NotifyWithParameterObjectAsync("textDocument/publishDiagnostics", Arg.Any<object>()))
-                .Do(x => {
-                    var dp = x.Args()[1] as PublishDiagnosticsParams;
-                    dp.Should().NotBeNull();
-                    dp.uri.Should().Be(uri);
-                    dp.diagnostics.Length.Should().Be(expectedSeverity == DiagnosticSeverity.Unspecified ? 0 : 1);
-                    if (expectedSeverity != DiagnosticSeverity.Unspecified) {
-                        dp.diagnostics[0].severity.Should().Be(expectedSeverity);
-                    }
-                    callReceived = true;
-                });
+                .Do(x => reported.Add(x.Args()[1] as PublishDiagnosticsParams));
 
             ds.DiagnosticsSeverityMap = new DiagnosticsSeverityMap(new[] { ErrorCodes.UnresolvedImport }, null, null, null);
-            callReceived.Should().BeTrue();
+            reported.Count.Should().Be(1);
+            reported[0].uri.Should().Be(doc.Uri);
+            reported[0].diagnostics.Length.Should().Be(1);
+            reported[0].diagnostics[0].severity.Should().Be(DiagnosticSeverity.Error);
 
-            expectedSeverity = DiagnosticSeverity.Information;
-            callReceived = false;
+            reported.Clear();
             ds.DiagnosticsSeverityMap = new DiagnosticsSeverityMap(null, null, new[] { ErrorCodes.UnresolvedImport }, null);
-            ds.Diagnostics[uri][0].Severity.Should().Be(Severity.Information);
-            callReceived.Should().BeTrue();
+            reported.Count.Should().Be(1);
+            reported[0].diagnostics[0].severity.Should().Be(DiagnosticSeverity.Information);
 
-            expectedSeverity = DiagnosticSeverity.Unspecified;
-            callReceived = false;
+            reported.Clear();
             ds.DiagnosticsSeverityMap = new DiagnosticsSeverityMap(null, null, null, new[] { ErrorCodes.UnresolvedImport });
-            ds.Diagnostics[uri].Count.Should().Be(0);
-            callReceived.Should().BeTrue();
-
-            expectedSeverity = DiagnosticSeverity.Unspecified;
-            callReceived = false;
-            ds.DiagnosticsSeverityMap = new DiagnosticsSeverityMap(new[] { ErrorCodes.UnresolvedImport }, null, null, new[] { ErrorCodes.UnresolvedImport });
-            ds.Diagnostics[uri].Count.Should().Be(0);
-            callReceived.Should().BeTrue();
+            reported.Count.Should().Be(1);
+            reported[0].diagnostics.Length.Should().Be(0);
         }
 
         [TestMethod, Priority(0)]
-        public async Task SuppressError() {
-            const string code = @"import nonexistent";
+        public async Task OnlyPublishChangedFile() {
```

Comment:
This test fails for me:

```
Test Name:	OnlyPublishChangedFile
Test FullName:	Microsoft.Python.LanguageServer.Tests.DiagnosticsTests.OnlyPublishChangedFile
Test Source:	C:\Users\jabaile\mikhail-language-server\src\LanguageServer\Test\DiagnosticsTests.cs : line 199
Test Outcome:	Failed
Test Duration:	0:00:04.0448316

Result StackTrace:	
at FluentAssertions.Execution.LateBoundTestFramework.Throw(String message) in C:\projects\fluentassertions-vf06b\Src\FluentAssertions\Execution\LateBoundTestFramework.cs:line 16
   at FluentAssertions.Execution.AssertionScope.FailWith(String message, Object[] args) in C:\projects\fluentassertions-vf06b\Src\FluentAssertions\Execution\AssertionScope.cs:line 225
   at FluentAssertions.Numeric.NumericAssertions`1.Be(T expected, String because, Object[] becauseArgs) in C:\projects\fluentassertions-vf06b\Src\FluentAssertions\Numeric\NumericAssertions.cs:line 50
   at Microsoft.Python.LanguageServer.Tests.DiagnosticsTests.OnlyPublishChangedFile() in C:\Users\jabaile\mikhail-language-server\src\LanguageServer\Test\DiagnosticsTests.cs:line 247
Result Message:	Expected ds.Diagnostics[doc1.Uri].Count to be 1, but found 0.
```

Reply:
Yes, added 
```cs
            await doc1.GetAstAsync();
            await doc1.GetAnalysisAsync(10000);
```
as above. This is the way it has to be done for now.